lib/twingly/url.rb

```diff
@@ -193,6 +200,14 @@ def <=>(other)
       self.to_s <=> other.to_s
     end
 
+    def eql?(other)
+      self.hash == other.hash
+    end
+
+    def hash
+      SHA256_DIGEST.digest(self.to_s).unpack1("q")
```

Comment:
This looks inspired by something, if so, could you share what?

Reply:
The autoping hash? :)

https://github.com/twingly/twingly-url/blob/65c09ae651d6752cd0f53e2def9201b259eb86d1/lib/twingly/url/hasher.rb#L30-L32

Reply:
Oh, yeah maybe :)

Reply:
Yeah, exatly. But I didn’t want to use our hasher, since it’s specific to our databases. 

I read the docs for eql? and hash, wich said there should be a deterministic number. Can’t find the docs from my phone right now though. But I basically just implemented it like we usually calculate hashes. 😀

Reply:
I guess we could use the string hash directly.

Reply:
New implementation in 3ecf26fc5d626d8ce53ab20a1b8b7366dfab09ec